csv.h

```diff
@@ -882,53 +883,53 @@ void parse(char *col, char *&x) {
 }
 
 template <class overflow_policy, class T>
-void parse_unsigned_integer(const char *col, T &x) {
-  x = 0;
+void parse_unsigned_integer(const char *col, optional<T> &x) {
+  x = {};
   while (*col != '\0') {
     if ('0' <= *col && *col <= '9') {
       T y = *col - '0';
-      if (x > (std::numeric_limits<T>::max() - y) / 10) {
-        overflow_policy::on_overflow(x);
+      if (x && *x > (std::numeric_limits<T>::max() - y) / 10) {
+        overflow_policy::on_overflow(*x);
         return;
       }
-      x = 10 * x + y;
+      x = (x ? 10 * *x : 0) + y;
     } else
       throw error::no_digit();
     ++col;
   }
 }
 
 template <class overflow_policy>
-void parse(char *col, uint8_t &x) {
+void parse(char *col, optional<uint8_t> &x) {
   parse_unsigned_integer<overflow_policy>(col, x);
 }
 template <class overflow_policy>
-void parse(char *col, uint16_t &x) {
+void parse(char *col, optional<uint16_t> &x) {
   parse_unsigned_integer<overflow_policy>(col, x);
 }
 template <class overflow_policy>
-void parse(char *col, uint32_t &x) {
+void parse(char *col, optional<uint32_t> &x) {
   parse_unsigned_integer<overflow_policy>(col, x);
 }
 template <class overflow_policy>
-void parse(char *col, uint64_t &x) {
+void parse(char *col, optional<uint64_t> &x) {
   parse_unsigned_integer<overflow_policy>(col, x);
 }
 
 template <class overflow_policy, class T>
-void parse_signed_integer(const char *col, T &x) {
+void parse_signed_integer(const char *col, optional<T> &x) {
   if (*col == '-') {
     ++col;
 
-    x = 0;
+    x = {};
     while (*col != '\0') {
       if ('0' <= *col && *col <= '9') {
         T y = *col - '0';
-        if (x < (std::numeric_limits<T>::min() + y) / 10) {
-          overflow_policy::on_underflow(x);
+        if (x && *x < (std::numeric_limits<T>::min() + y) / 10) {
+          overflow_policy::on_underflow(*x);
           return;
         }
-        x = 10 * x - y;
+        *x = (x ? 10 * *x : 0) - y;
```

Comment:
I think you meant this? 
```suggestion
        x = (x ? 10 * *x : 0) - y;
```

Reply:
Indeed, thanks!